src/jsdoc_transformer.ts

```diff
@@ -112,88 +112,107 @@ export function maybeAddHeritageClauses(
-      // that is a TS-specific thing and we need to translate it to the
-      // the Closure "@implements {Foo}".
+  for (const heritage of decl.heritageClauses) {
+    const isExtends = heritage.token === ts.SyntaxKind.ExtendsKeyword;
+    if (isClass && isExtends) {
+      // If a class has an "extends", that is preserved in the ES6 output
+      // and we don't need to emit any additional jsdoc.
+      //
       // However for ambient declarations, we only emit externs, and in those we do need to
       // add "@extends {Foo}" as they use ES5 syntax.
-      continue;
+      if (!isAmbient(decl)) continue;
     }
-    for (const impl of heritage.types) {
-      let tagName = decl.kind === ts.SyntaxKind.InterfaceDeclaration ? 'extends' : 'implements';
-
-      const sym = mtt.typeChecker.getSymbolAtLocation(impl.expression);
-      if (!sym) {
-        // It's possible for a class declaration to extend an expression that
-        // does not have have a symbol, for example when a mixin function is
-        // used to build a base class, as in `declare MyClass extends
-        // MyMixin(MyBaseClass)`.
-        //
-        // Handling this correctly is tricky. Closure throws on this
-        // `extends <expression>` syntax (see
-        // https://github.com/google/closure-compiler/issues/2182). We would
-        // probably need to generate an intermediate class declaration and
-        // extend that. For now, just omit the `extends` annotation.
-        mtt.debugWarn(decl, `could not resolve supertype: ${impl.getText()}`);
-        docTags.push({
-          tagName: '',
-          text: 'NOTE: tsickle could not resolve supertype, ' +
-              'class definition may be incomplete.\n'
-        });
+
+    // Otherwise, if we get here, we need to emit some jsdoc.
+    for (const expr of heritage.types) {
+      const parentName = heritageName(isExtends, expr);
+      // parentName may be null, indicating that the clause is something inexpressible
+      // in Closure, e.g. "class Foo implements Partial<Bar>".  For 'extends' clauses
+      // that means we cannot emit anything at all.
+      if (isExtends && !parentName) {
         continue;
       }
-      let alias: ts.Symbol = sym;
-      if (sym.flags & ts.SymbolFlags.TypeAlias) {
-        // It's implementing a type alias.  Follow the type alias back
-        // to the original symbol to check whether it's a type or a value.
-        const type = mtt.typeChecker.getDeclaredTypeOfSymbol(sym);
-        if (!type.symbol) {
-          // It's not clear when this can happen, but if it does all we
-          // do is fail to emit the @implements, which isn't so harmful.
-          continue;
-        }
-        alias = type.symbol;
-      }
-      if (alias.flags & ts.SymbolFlags.Alias) {
-        alias = mtt.typeChecker.getAliasedSymbol(alias);
+      docTags.push({
+        tagName: isExtends ? 'extends' : 'implements',
+        type: parentName || 'InexpressibleType',
```

Comment:
Can you add a bit more commentary on what `InexpressibleType` should achieve on the Closure side? Why do we prefer emitting that over just not emitting anything?

Reply:
I think the earlier comment now covers it.